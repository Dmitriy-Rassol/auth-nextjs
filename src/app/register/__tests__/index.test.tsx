import mockRouter from "next-router-mock";
import { fireEvent, getByTestId, render, screen } from "@testing-library/react";
import Register from "../page";
import "@testing-library/jest-dom";

const setup = () => {
  const handler = jest.fn();

  const utils = render(<Register />);
  // Assertion, так как мы уверенны, что здесь будет input-элемент,
  // а тип по умолчанию HTMLElement не содержит свойства value, необходимого нам в 3м тесте

  return {
    handler,
    ...utils,
  };
};

// jest.mock('next/navigation', () => require('next-router-mock'));
jest.mock("next/navigation", () => {
  return {
    useRouter: () => {},
  };
});

describe("Register Page", () => {
  it("should match snapshot", () => {
    const { container } = setup();

    expect(container).toMatchSnapshot();
  });

  it("displays error message for required login field on blur and removes it on input", () => {
    render(<Register />);

    const loginInput = screen.getByPlaceholderText("Your login");

    // Эмулируем событие Blur на поле Login
    fireEvent.blur(loginInput);

    // Проверяем, что текст ошибки присутствует в документе
    expect(
      screen.getByText("The login field is required.")
    ).toBeInTheDocument();

    // Эмулируем ввод текста в поле Login
    fireEvent.change(loginInput, { target: { value: "testlogin" } });

    // Проверяем, что текст ошибки более не присутствует в документе
    expect(
      screen.queryByText("The login field is required.")
    ).not.toBeInTheDocument();
  });
});
