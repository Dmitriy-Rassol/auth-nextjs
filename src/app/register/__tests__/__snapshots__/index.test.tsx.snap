// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Register Page should match snapshot 1`] = `
<div>
  <form
    autocomplete="off"
    method="POST"
  >
    <div
      class="mt-8"
    >
      <label
        class="block mb-2 text-sm font-medium text-gray-900 dark:text-white"
      >
        Your login
      </label>
      <input
        aria-label="Your login"
        class="bg-gray-50 border border-gray-300 text-gray-900 sm:text-sm rounded-lg focus:ring-primary-600 focus:border-primary-600 block w-full p-2.5 dark:bg-gray-700 dark:border-gray-600 dark:placeholder-gray-400 dark:text-white dark:focus:ring-blue-500 dark:focus:border-blue-500"
        id="login_Your login"
        name="login"
        placeholder="Your login"
        required=""
        type="text"
        value=""
      />
    </div>
    <p
      style="display: none;"
    />
    <div
      class="mt-8"
    >
      <label
        class="block mb-2 text-sm font-medium text-gray-900 dark:text-white"
      >
        Your email
      </label>
      <input
        aria-label="Your email"
        class="bg-gray-50 border border-gray-300 text-gray-900 sm:text-sm rounded-lg focus:ring-primary-600 focus:border-primary-600 block w-full p-2.5 dark:bg-gray-700 dark:border-gray-600 dark:placeholder-gray-400 dark:text-white dark:focus:ring-blue-500 dark:focus:border-blue-500"
        id="email_Your email"
        name="email"
        placeholder="Your email"
        required=""
        type="email"
        value=""
      />
    </div>
    <p
      style="display: none;"
    />
    <div
      class="mt-8"
    >
      <label
        class="block mb-2 text-sm font-medium text-gray-900 dark:text-white"
      >
        Your password
      </label>
      <input
        aria-label="Your password"
        class="bg-gray-50 border border-gray-300 text-gray-900 sm:text-sm rounded-lg focus:ring-primary-600 focus:border-primary-600 block w-full p-2.5 dark:bg-gray-700 dark:border-gray-600 dark:placeholder-gray-400 dark:text-white dark:focus:ring-blue-500 dark:focus:border-blue-500"
        id="password_Your password"
        name="password"
        placeholder="Your password"
        required=""
        type="password"
        value=""
      />
    </div>
    <p
      style="display: none;"
    />
    <div
      class="mt-8"
    >
      <button
        class="w-full text-white bg-blue-600 hover:bg-blue-700 focus:ring-4 focus:outline-none focus:ring-primary-300 font-medium rounded-lg text-sm px-5 py-2.5 text-center dark:bg-primary-600 dark:hover:bg-primary-700 dark:focus:ring-primary-800"
        disabled=""
        style="background: grey;"
        type="submit"
      >
        Sign up
      </button>
    </div>
  </form>
</div>
`;
